src/v/cluster/archival/ntp_archiver_service.cc

```diff
@@ -2340,18 +2341,16 @@ ntp_archiver::replicate_archival_metadata(
               fence.unsafe_add,
               _parent.archival_meta_stm()->manifest().get_applied_offset(),
               _parent.archival_meta_stm()->get_insync_offset());
-            batch_builder.read_write_fence(fence.read_write_fence);
+            rw_fence = fence.read_write_fence;
         }
-        batch_builder.add_segments(
+        auto error = co_await _parent.archival_meta_stm()->add_segments(
           meta,
-          checks_disabled ? cluster::segment_validated::no
-                          : cluster::segment_validated::yes);
-        if (manifest_clean_offset.has_value()) {
-            batch_builder.mark_clean(manifest_clean_offset.value());
-        }
-        batch_builder.update_highest_producer_id(highest_producer_id);
-
-        auto error = co_await batch_builder.replicate();
+          manifest_clean_offset,
+          highest_producer_id,
+          deadline,
+          _as,
+          is_validated,
+          rw_fence);
```

Comment:
> archival: Use archival_metadata_stm.add_segments method
Switch back to using 'add_segments' method

I don't understand the context for this change. Is this changing an "issue" with previous commits, or something net new?

Reply:
It stops using the command builder and moves back to using the `add_segments` which was modified by the previous commit.